src/basic/raw-clone.h

```diff
@@ -47,23 +48,28 @@ static inline pid_t raw_clone(unsigned long flags) {
                  * %o1. Inline assembly is needed to get the flag returned
                  * in %o1.
                  */
-                int in_child, child_pid;
+                int in_child, child_pid, error;
 
-                asm volatile("mov %2, %%g1\n\t"
-                             "mov %3, %%o0\n\t"
+                asm volatile("mov %3, %%g1\n\t"
+                             "mov %4, %%o0\n\t"
                              "mov 0 , %%o1\n\t"
 #if defined(__arch64__)
```

Comment:
Whoa, can `__arch64__` be even set if `__sparc__` is set? Isn't this `#if` dead code?

Reply:
I think you're mixing up `__aarch64__` (64-bit ARM) with `__arch64__` (generic 64-bit platform flag)...

Having said that, `__arch64__` looks fairly obscure. Don't know if there's a better way to do this.

Googling for it, I found [this](https://lists.debian.org/debian-sparc/2016/05/msg00011.html) which is in the context of SPARC, so maybe that's the way to do this on that platform...

I think I'd recommend leaving this one as-is, given this seems to be working (for the author.)

Reply:
The `__arch64__` isn't technically my code; I just moved it down from the top of the `__sparc__` block.

Anyway, I've been testing in a 64-bit sparc VM (qemu-system-sparc64), and this code definitely works on Debian sid. One of my peers also confirmed it works in a 32-bit userspace (Gentoo) on real hardware.

Reply:
Thanks @filbranden, I mixed up the two. I blame the arm holdings, they should have called the damn thing "arm64", would have been much easier for everybody ;)